### ISSUE TITLE
question about the loss function, threshold and pre-trained model

### ISSUE THREAD

**Issue by KoalaSheep** [Ien001/AG-CNN#1]
Thank you for your re-implementation. 
1. I found you use 
loss = loss1 * 0.8 + loss2 * 0.1 + loss3 * 0.1
to calculate the global branch, local branch and fusion branch bce loss. 
However, I can't find anything alike in the paper. 
It seems that author doesn't combine them as one loss to perform optimization. 
Please let me know your reason about this combination. 

2. I also notice in your Attention_gen_patchs, you use cv2.threshold instead of the given threshold(0.7) from the paper. 
Since your local branch mean AUC is lower than the paper, have you tried 0.7 as threshold to crop the patch?

3. thank you for your trained model. 
I notice that the global branch model is named as xxxx_epoch_1.pkl, 
and local branch model is named as xxx_epoch_2.pkl. 
As you save the model with 'str(epoch)', I am curious that, did you achieve the best mean AUC within only 1 or 2 epochs?

**Comment by Ien001** [comment]
> Thank you for your re-implementation.
> 
> 1. I found you use
>    loss = loss1 * 0.8 + loss2 * 0.1 + loss3 * 0.1
>    to calculate the global branch, local branch and fusion branch bce loss.
>    However, I can't find anything alike in the paper.
>    It seems that author doesn't combine them as one loss to perform optimization.
>    Please let me know your reason about this combination.
> 2. I also notice in your Attention_gen_patchs, you use cv2.threshold instead of the given threshold(0.7) from the paper.
>    Since your local branch mean AUC is lower than the paper, have you tried 0.7 as threshold to crop the patch?
> 3. thank you for your trained model.
>    I notice that the global branch model is named as xxxx_epoch_1.pkl,
>    and local branch model is named as xxx_epoch_2.pkl.
>    As you save the model with 'str(epoch)', I am curious that, did you achieve the best mean AUC within only 1 or 2 epochs?

Hi, sorry for the late reply.

Q1&Q2: 
They are empirical settings. 
`loss = loss1 * 0.8 + loss2 * 0.1 + loss3 * 0.1`
this setting would allow the three networks learn from all three losses during BP.
About the threshold, yes, I have tried 0.7, but only got worse results.

Q3:
Yes, G and L branch could achieve that at some early stages, because the models were pre-trained in Chest X-ray dataset.

**Comment by KoalaSheep** [comment]
Thank you for your reply.
